libr/core/cmd_anal.c

```diff
@@ -987,6 +988,41 @@ static void var_accesses_list(RAnalFunction *fcn, RAnalVar *var, int access_type
 	r_cons_newline ();
 }
 
+static void list_vars2(RCore *core, RAnalFunction *fcn, int type) {
+	RAnalVar *var;
+	RListIter *iter;
+
+	if (type == '*') {
+		RList *list = r_anal_var_all_list (core->anal, fcn);
+		const char *bp = r_reg_get_name (core->anal->reg, R_REG_NAME_BP);
+		r_cons_printf ("f-fcnvar*\n");
+		r_list_foreach (list, iter, var) {
+			r_cons_printf ("f fcnvar.%s @ %s%s%d\n", var->name, bp,
+				var->delta>=0? "+":"", var->delta);
+		}
+		return;
+	}
+	RList *list = r_anal_var_all_list (core->anal, fcn);
+	r_list_foreach (list, iter, var) {
+		r_cons_printf ("* %s\n", var->name);
+		RAnalVarAccess *acc;
+		r_vector_foreach (&var->accesses, acc) {
+			if (!(acc->type & R_ANAL_VAR_ACCESS_TYPE_READ)) {
+				continue;
+			}
+			r_cons_printf ("R 0x%"PFMT64x"  ", (ut64)((st64)fcn->addr + acc->offset));
+			r_core_cmdf (core, "pi 1 @ 0x%08"PFMT64x, fcn->addr + acc->offset);
```

Comment:
for this, maybe you can use `r_core_print_disasm_instructions` instead.

Reply:
ideally no core.cmd()  calls  should be done  inside r2, so yeah  i would change this when i have some spare time